cdc/scheduler/internal/v3/agent/agent.go

```diff
@@ -250,8 +253,9 @@ func (a *agent) handleMessage(msg []*schedulepb.Message) []*schedulepb.Message {
 
 		switch message.GetMsgType() {
 		case schedulepb.MsgHeartbeat:
-			response := a.handleMessageHeartbeat(message.GetHeartbeat())
-			result = append(result, response)
+			var reMsg *schedulepb.Message
+			reMsg, barrier = a.handleMessageHeartbeat(message.GetHeartbeat())
+			result = append(result, reMsg)
```

Comment:
`Barrier` contains both `globalBarrierTs` and `tableBarrier`. If there is no any tables that have DDL, `tableBarrier` will be nil, but `globalBarrierTs` need to be received from every owner heartbeat since the `status.ResolvedTs` is no longer works as a barrier.